host/xtest/pkcs11_1000.c

```diff
@@ -1463,3 +1483,492 @@ static void xtest_pkcs11_test_1007(ADBG_Case_t *c)
+
+static CK_ATTRIBUTE cktest_hmac_sha256_key2[] =
+	CK_MAC_KEY_HMAC(CKK_SHA256_HMAC, mac_data_sha256_key2);
+
+static CK_ATTRIBUTE cktest_hmac_sha384_key[] =
+	CK_MAC_KEY_HMAC(CKK_SHA384_HMAC, mac_data_sha384_key1);
+
+static CK_ATTRIBUTE cktest_hmac_sha512_key[] =
+	CK_MAC_KEY_HMAC(CKK_SHA512_HMAC, mac_data_sha512_key1);
+
+struct mac_test {
+	CK_ATTRIBUTE_PTR attr_key;
+	CK_ULONG attr_count;
+	CK_MECHANISM_PTR mechanism;
+	size_t in_incr;
+	const uint8_t *in;
+	size_t in_len;
+	const uint8_t *out;
+	size_t out_len;
+	bool multiple_incr;
+};
+
+#define CKTEST_MAC_TEST(key, mecha, input_incr, input, output, incr) {	\
+		.attr_key = key,		\
+		.attr_count = ARRAY_SIZE(key),	\
+		.mechanism = mecha,		\
+		.in_incr = input_incr,		\
+		.in = input,				\
+		.in_len = ARRAY_SIZE(input),		\
+		.out = output,				\
+		.out_len = ARRAY_SIZE(output),		\
+		.multiple_incr = incr			\
+	}
+
+static const struct mac_test cktest_mac_cases[] = {
+	CKTEST_MAC_TEST(cktest_hmac_md5_key, &cktest_hmac_md5_mechanism,
+			4, mac_data_md5_in1, mac_data_md5_out1, false),
+	CKTEST_MAC_TEST(cktest_hmac_sha1_key, &cktest_hmac_sha1_mechanism,
+			5, mac_data_sha1_in1, mac_data_sha1_out1, false),
+	CKTEST_MAC_TEST(cktest_hmac_sha224_key, &cktest_hmac_sha224_mechanism,
+			8, mac_data_sha224_in1, mac_data_sha224_out1, false),
+	CKTEST_MAC_TEST(cktest_hmac_sha256_key1, &cktest_hmac_sha256_mechanism,
+			1, mac_data_sha256_in1, mac_data_sha256_out1, false),
+	CKTEST_MAC_TEST(cktest_hmac_sha256_key2, &cktest_hmac_sha256_mechanism,
+			7, mac_data_sha256_in2, mac_data_sha256_out2, false),
+	CKTEST_MAC_TEST(cktest_hmac_sha384_key, &cktest_hmac_sha384_mechanism,
+			11, mac_data_sha384_in1, mac_data_sha384_out1, false),
+	CKTEST_MAC_TEST(cktest_hmac_sha512_key, &cktest_hmac_sha512_mechanism,
+			13, mac_data_sha512_in1, mac_data_sha512_out1, false),
+};
+
+static void xtest_pkcs11_test_1008(ADBG_Case_t *c)
+{
+	CK_RV rv = CKR_GENERAL_ERROR;
+	CK_SLOT_ID slot = 0;
+	CK_SESSION_HANDLE session = CK_INVALID_HANDLE;
+	CK_FLAGS session_flags = CKF_SERIAL_SESSION;
+	CK_OBJECT_HANDLE key_handle = CK_INVALID_HANDLE;
+	uint8_t out[512] = { 0 };
+	CK_ULONG out_size = 0;
+	int close_subcase = 0;
+	int destroy_object = 0;
+	struct mac_test const *test = NULL;
+	size_t n = 0;
+
+	rv = init_lib_and_find_token_slot(&slot);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		return;
+
+	rv = C_OpenSession(slot, session_flags, NULL, 0, &session);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		return;
+
+	for (n = 0; n < ARRAY_SIZE(cktest_mac_cases); n++) {
+
+		test = &cktest_mac_cases[n];
+		Do_ADBG_BeginSubCase(c, "Sign case %zu algo (%s)",
+				     n,
+				     ckm2str(test->mechanism->mechanism));
+
+		close_subcase = 1;
+		destroy_object = 1;
+
+		rv = C_CreateObject(session, test->attr_key, test->attr_count,
+				    &key_handle);
+
+		if (!ADBG_EXPECT_CK_OK(c, rv))
+			goto out;
+
+		/* Test signature in 1 step */
+		if (test->in != NULL) {
+			rv = C_SignInit(session, test->mechanism, key_handle);
+
```

Comment:
I think the empty lines between `C_*()` calls and return value test could be removed. There were originally intentionally added to highlight cryptoki API function calls but in the end, I don't think there're that useful.

Reply:
Removed the empty line throughout